SNT/TypeGenerator.tt

```diff
@@ -0,0 +1,157 @@
+<#@ template language="C#" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Collections.Generic" #>
+using System;
+using System.Linq;
+using System.Numerics.Tensors;
+using TorchSharp;
+
+namespace Torch.SNT 
+{
+<#
+foreach (var type in new [] { 
+    new { ApiName = "Byte", Storage = "byte", AccReal="long" }, 
+    new { ApiName = "Short", Storage = "short", AccReal="long" },   
+    new { ApiName = "Int", Storage = "int",  AccReal="long"},
+    new { ApiName = "Long", Storage = "long", AccReal="long" },
+    new { ApiName = "Double", Storage = "double", AccReal="double" },
+    new { ApiName = "Float", Storage = "float", AccReal="double" },
+     }) {
+        string tname = type.ApiName;
+#>
+    public sealed class <#=tname#>TorchTensor : DenseTensor<<#=tname.ToLower()#>>
```

Comment:
Even if we don't split up the Razor template file, maybe we should avoid adding new logic to the existing one?

Reply:
I didn't get this. This TypeGenerator is different than the one in TorchSharp and is used to generate SNT wrappers around the original torch tensors.

Reply:
Aha! I didn't pay attention to the folder name.